Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/ExpLog.lean

```diff
@@ -0,0 +1,194 @@
+/-
+Copyright (c) 2024 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import Mathlib.Analysis.NormedSpace.Spectrum
+import Mathlib.Analysis.SpecialFunctions.Exponential
+import Mathlib.Topology.ContinuousFunction.FunctionalCalculus
+
+/-!
+# The exp and log functions based on the continuous functional calculus
+
+This file defines the log function via the continuous functional calculus and build its API.
+This allows one to take logs of matrices, operators, elements of a C⋆-algebra, etc.
+
+It also shows that exponentials defined via the CFC are equal to `NormedSpace.exp` (defined via
+power series) whenever the former are not junk values.
+
+## Main declarations
+
++ `CFC.log`: the real log function based on the CFC, i.e. `cfc Real.log`
++ `CFC.exp_eq_normedSpace_exp`: exponentials based on the CFC are equal to exponentials based
+  on power series.
++ `CFC.log_exp` and `CFC.exp_log`: `CFC.log` and `NormedSpace.exp ℝ` are inverses of each other.
+
+## Implementation notes
+
+Since `cfc Real.exp` and `cfc Complex.exp` are strictly less general than `NormedSpace.exp`
+(defined via power series), we only give minimal API for these here in order to relate
+`NormedSpace.exp` to functions defined via the CFC. In particular, we don't give separate
+definitions for them.
+
+## TODO
+
++ Show that `log (a * b) = log a + log b` whenever `a` and `b` commute (and the same for indexed
+  products).
++ Relate `CFC.log` to `rpow`, `zpow`, `sqrt`, `inv`.
+-/
+
+open NormedSpace
+
+section general_exponential
+variable {𝕜 : Type*} {α : Type*} [RCLike 𝕜] [TopologicalSpace α] [CompactSpace α]
+
+lemma NormedSpace.exp_continuousMap_eq (f : C(α, 𝕜)) :
+    exp 𝕜 f = (⟨exp 𝕜 ∘ f, Continuous.comp exp_continuous f.continuous⟩ : C(α, 𝕜)) := by
```

Comment:
It feels like this lemma should belong with `NormedSpace.exp`, but none of those files have the right imports. `#find_home` tells me to put it in `Mathlib.Analysis.NormedSpace.Spectrum`, but that feels even more out of place than just leaving it here.